### PR TITLE
Make the frameProcessor shutdown command-driven

### PR DESCRIPTION
**Pull request by timcnicholls**
This PR address #153 , switching the frameProcessor shutdown mechanism command rather than configuration driven.

### CODE REVIEW

**Comment by timcnicholls** [comment]
@ulrikpedersen It's just an IpcMessage change from a `config` message with a `shutdown` param message explicitly present to an explicit `cmd` type message. I'm not actually certain how much this is used beyond the particular use case (LPD) that raised the issue?

**Comment by GDYendell** [comment]
Does the adapter need changes to expose this new API? Currently I do 

`requests.put("fp/config/shutdown", True)`

to the adapter and it just passes the message straight through.

**Comment by timcnicholls** [comment]
I _think_ it should be sufficient to replace this with `requests.put("fp/command/shutdown")` ? @ajgdls will know better. Does it need adding to `SUPPORTED_COMMANDS`?

**Comment by ajgdls** [comment]
Yes it will need adding to the supported commands and the replacement that @timcnicholls suggested should work.

**Comment by timcnicholls** [comment]
Thanks @ajgdls I'll add this to the PR and check it works OK

**Comment by timcnicholls** [comment]
@GDYendell Latest push works fine with command-line equivalent:
```
curl -H "Content-Type: application/json" -X PUT -d '' http://127.0.0.1:8888/api/0.1/fp/command/shutdown
```
and likewise for the FR.

File: tools/python/odin_data/odin_data_adapter.py [hunk 1]
```diff
@@ -22,9 +22,10 @@ class OdinDataAdapter(ApiAdapter):
     ERROR_NO_RESPONSE = "No response from client, check it is running"
     ERROR_FAILED_TO_SEND = "Unable to successfully send request to client"
     ERROR_FAILED_GET = "Unable to successfully complete the GET request"
+    ERROR_FAILED_PUT = "Unable to successfully complete the PUT request"
     ERROR_PUT_MISMATCH = "The size of parameter array does not match the number of clients"
 
-    SUPPORTED_COMMANDS = ['reset_statistics', 'request_version']
+    SUPPORTED_COMMANDS = ['reset_statistics', 'request_version', 'shutdown']
```
[the remark this answers is not in the document]
Comment (ajgdls) [reply tools/python/odin_data/odin_data_adapter.py]: The request configuration and status commands are automatically sent by the adapter itself, and the information made available through a GET.  This was how the original adapter code worked.  The commands could be added to the supported commands, there would be no positive or negative side effects that I could see.

[the remark this answers is not in the document]
Comment (MattTaylorDLS) [reply tools/python/odin_data/odin_data_adapter.py]: ah ok cheers

**Comment by timcnicholls** [comment]
@GDYendell Are you happy with this?

**Pull request merged**